src/python/pants/jvm/target_types.py

```diff
@@ -102,6 +102,7 @@ class JvmArtifactUrlField(StringField):
 class JvmArtifactJarSourceField(SingleSourceField):
     alias = "jar"
     expected_file_extensions = (".jar",)
+    expected_num_files = range(0, 2)
```

Comment:
It feels like this is still a gap in the source validation facility (for the reason mentioned on the ticket), but a warning should be sufficient here.